src/main/kotlin/org/wycliffeassociates/otter/common/domain/content/Recordable.kt

```diff
@@ -0,0 +1,54 @@
+package org.wycliffeassociates.otter.common.domain.content
+
+import org.wycliffeassociates.otter.common.data.model.ContentType
+import org.wycliffeassociates.otter.common.data.workbook.*
+
+class Recordable private constructor(
+    val textItem: TextItem? = null,
```

Comment:
This is nullable since Chunk's textItem is nullable (which in turn is because Content's text column is nullable.)
This feels bad, and it seems like we shouldn't even be able to create a Recordable if the text is null

Reply:
I think I agree.  Does it make sense to just make this non-nullable and have build() return a nullable?

Reply:
Perhaps, but if we just return a null Recordable, we still wouldn't know what happened. Should we throw an exception, or perhaps use logging to log what happened? We don't seem to do much of either of these options in the current app.

Reply:
Let's talk about the purpose of this object; textItem may not make sense to be owned by it?

Reply:
Recordable is meant to be a generalized object for something that can be recorded, be it scripture or resources. It should have the information needed to display the text to record, and to generate a wav file (filename and all) when the user records a take. That's my idea for it anyway

Reply:
I think it should probably be focused on just information just needed to create a recording, as you can see textItem already starts to introduce behavior that some recordables might have and others don't.

Recordable could be an interface maybe?

Reply:
Hm, the concept of a Recordable that doesn't have a text item doesn't make sense to me- what are you recording then? I only had to make textItem nullable because it can propagate from the fact that Content's text column is nullable (but I don't think we'd ever want to make a Recordable based on such a Content with a null text field.)

I'm not saying there definitely isn't a better way to do this, but if you take a look at how I'm using building the Recordables in ResourcesViewModel and using them in the TakesViewModel, I wonder if you would agree or disagree with the approach.

Reply:
> Hm, the concept of a Recordable that doesn't have a text item doesn't make sense to me- what are you recording then?

What is the purpose of the apps that we're building? Similarly:

> I don't think we'd ever want to make a Recordable based on such a Content with a null text field.

I do

> I'm not saying there definitely isn't a better way to do this, but if you take a look at how I'm building the Recordables in ResourcesViewModel and using them in TakesViewModel, I wonder if you would agree or disagree with the approach.

Reply:
Updating this thread after my conversation with Joe yesterday: 

Oral-to-oral translations are a case where the `text` field will be null. Drawing from the single responsibility principle, we are looking at creating an interface/class whose job is simply to name an audio file. The old `Recordable` object will either implement this interface or contain this "namer" object. Then we will handle the separate use cases of text-to-audio or audio-to-audio, perhaps by subclassing Recordable.